feat(toolkits/pro): add package name and description

fengyon opened the pull request:
1. add name of package from the dirctory where the command was executed; 
2. add description of package from the input of command

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-cli/blob/main/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

执行  `tiny init vue-pro` 初始化 TinyVue项目
执行 `tiny init ng-pro`  初始化 TinyNG项目

<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: N/A

## What is the new behavior?

执行 `tiny init pro`  初始化 TinyNG项目
例如在 D;/test执行目录
执行后：
  1、请输入项目名称： (test)            // 默认当前目录
  2、请输入项目描述： （基于TinyPro套件创建的中后台系统）
  3、请选择你希望使用的技术栈： (Use arrow keys)
       > vue
          angular
  4、开始初始化

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->

## Other information

h-ivy commented on packages/cli/module-list/package.json:
```diff
@@ -1,12 +1,11 @@
     "@opentiny/tiny-plugin-hwc": "快速管理和访问华为云服务资源",
-    "@opentiny/tiny-toolkit-ng-pro": "基于TinyNG的开箱即用前端解决方案",
-    "@opentiny/tiny-toolkit-vue-pro": "基于TinyVue的开箱即用前端解决方案",
+    "@opentiny/tiny-toolkit-pro": "基于TinyUI的开箱即用前端解决方案",
```
要不改成这个，开箱即用的中后台前端/设计解决方案

fengyon replied:
已修改

h-ivy commented on packages/toolkits/pro/src/lib/init.ts:
```diff
@@ -2,19 +2,63 @@ import * as path from 'path';
-import inquirer from 'inquirer';
+import inquirer, { QuestionCollection } from 'inquirer';
 import { cliConfig, logs, fs, user, modules } from '@opentiny/cli-devkit';
+import * as fsNode from 'fs';
```
1、import的顺序稍微统一一下，比如可以先import第三方的包、再import自研的包，最后再import相对路径的依赖，其它文件也都检查一下，按照这个顺序
2、@opentiny/cli-devkit有fs了，不需要在第七行再引入fs

fengyon replied:
已修改

h-ivy commented on packages/toolkits/pro/src/lib/init.ts:
```diff
@@ -2,19 +2,63 @@ import * as path from 'path';
+const getInitAnswers = (): Promise<InitAnswers> => {
+  const basename = path.basename(utils.getDistPath());
+
+  const question: QuestionCollection<InitAnswers> = [
```
21和23行之间为啥留个空行，一般所有的const定义会和逻辑中间留一个空行

fengyon replied:
已删除此空行

h-ivy commented on packages/toolkits/pro/src/lib/init.ts:
```diff
@@ -2,19 +2,63 @@ import * as path from 'path';
+      type: 'input',
+      name: 'description',
+      message: '请输入项目描述：',
+      default: 'Based on @opentiny/tiny-toolkit-pro',
```
项目描述默认值为啥是Based on @opentiny/tiny-toolkit-pro，感觉这个默认值怪怪的，比如改成xxx中后台系统

fengyon replied:
已经改成 `基于TinyPro套件创建的中后台系统`

h-ivy commented on packages/toolkits/pro/src/lib/init.ts:
```diff
@@ -2,19 +2,63 @@ import * as path from 'path';
+    {
+      type: 'list',
+      name: 'framework',
+      message: '请选择你希望使用的技术栈：',
```
你改成您

fengyon replied:
已修改

h-ivy commented on packages/toolkits/pro/src/lib/init.ts:
```diff
@@ -60,18 +95,36 @@ export default async () => {
+    const writeOrReadOptions = { encoding: 'utf8' } as const;
+
+    const packageJson = JSON.parse(
+      fsNode.readFileSync(packageJsonPath, writeOrReadOptions)
```
用@opentiny/cli-devkit里面的fs提供的文件方法

fengyon replied:
已修改

h-ivy commented on packages/toolkits/pro/src/lib/init.ts:
```diff
@@ -102,5 +155,26 @@ export default async () => {
       '\n-------------------- 技术支持：官方小助手微信opentiny-official --------------------\n'
```
技术支持：官方小助手微信opentiny-official，有这玩意吗，找一下曾令卡确认一下

fengyon replied:
确认过了，有的

Commit:
feat(toolkits/pro): add package name and description

1. add name of package from the dirctory where the command was executed;
2. add description of
package from the input of command

Commit:
feat(toolkits/pro): add package name and description

1. add name of package from the dirctory where the command was executed;
2. add description of
package from the input of command

h-ivy commented on packages/cli/module-list/package.json:
```diff
@@ -1,12 +1,11 @@
     "@opentiny/tiny-plugin-hwc": "快速管理和访问华为云服务资源",
-    "@opentiny/tiny-toolkit-ng-pro": "基于TinyNG的开箱即用前端解决方案",
-    "@opentiny/tiny-toolkit-vue-pro": "基于TinyVue的开箱即用前端解决方案",
+    "@opentiny/tiny-toolkit-pro": "基于TinyUI的开箱即用前端解决方案",
```
要不改成这个，开箱即用的中后台前端/设计解决方案

fengyon replied:
已修改

h-ivy commented on packages/cli/module-list/package.json:
```diff
@@ -1,12 +1,11 @@
     "@opentiny/tiny-plugin-hwc": "快速管理和访问华为云服务资源",
-    "@opentiny/tiny-toolkit-ng-pro": "基于TinyNG的开箱即用前端解决方案",
-    "@opentiny/tiny-toolkit-vue-pro": "基于TinyVue的开箱即用前端解决方案",
+    "@opentiny/tiny-toolkit-pro": "基于TinyUI的开箱即用前端解决方案",
```
要不改成这个，开箱即用的中后台前端/设计解决方案

fengyon replied:
comment test

fengyon replied:
comment test

Commit:
feat(toolkits/pro): add package name and description

1. add name of package from the dirctory where the command was executed;
2. add description of
package from the input of command

fengyon reviewed (commented):
test

fengyon commented on packages/toolkits/pro/src/lib/interfaces.ts:
```diff
@@ -5,3 +5,12 @@ export interface CliOption {
+/**
+ * 初始化问题的选项 -> 选择的值
+ */
+export interface InitAnswers {
```
test comment

Commit:
feat(toolkits/pro): add package name and description

1. add name of package from the dirctory where the command was executed;
2. add description of
package from the input of command
